File: app/src/main/java/me/pwcong/jpstart/adapter/BannerPagerAdapter.java
package me.pwcong.jpstart.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import java.util.List;

import me.pwcong.jpstart.component.fragment.BannerFragment;
import me.pwcong.jpstart.mvp.bean.BannerItem;

/**
 * Created by Pwcong on 2016/10/10.
 */

public class BannerPagerAdapter extends BasePagerAdapter<BannerItem> {

    public BannerPagerAdapter(FragmentManager fm, List<BannerItem> list) {
        super(fm, list);
    }

    @Override
    protected Fragment getFragment(BannerItem bannerItem) {
        return BannerFragment.newInstance(bannerItem);
    }

    @Override
    protected CharSequence getTitle(BannerItem bannerItem) {
        return bannerItem.getTitle();
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/adapter/BannerPagerAdapter.kt
package com.github.pwcong.jpstart.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.github.pwcong.jpstart.mvp.bean.BannerItem
import com.github.pwcong.jpstart.ui.fragment.BannerFragment

class BannerPagerAdapter(fm: FragmentManager, list: List<BannerItem>) :
    BasePagerAdapter<BannerItem>(fm, list) {
    override fun getFragment(t: BannerItem): Fragment {
        return BannerFragment.newInstance(t)
    }

    override fun getTitle(t: BannerItem): CharSequence? {
        return t.title
    }
}
